pkg/storage/stores/series/series_index_gateway_store.go

```diff
@@ -88,7 +88,7 @@ func (c *IndexGatewayClientStore) LabelValuesForMetricName(ctx context.Context,
 	})
 	if isUnimplementedCallError(err) {
 		// Handle communication with older index gateways gracefully, by falling back to the index store calls.
-		return c.IndexStore.LabelNamesForMetricName(ctx, userID, from, through, metricName)
+		return c.IndexStore.LabelValuesForMetricName(ctx, userID, from, through, metricName, labelName, matchers...)
```

Comment:
is it tested?

Reply:
No it is not currently, but I have #5968 prepared, that covers this aspect.

Reply:
I'll try to find a way to test this without a huge amount of boilerplate code.